hell/adventcalendar/2025/5/index.md

```diff
@@ -1,35 +1,256 @@
 ---
 title: "Idrefs: What they are and how to use them"
-author: "Your Name"
-author_bio: "Your short bio"
+author: "Kilian Valkhof"
+author_bio: "Web developer and creator of Polypane.app, the browser for developers."
 date: 2025-12-05
 author_links:
-  - label: "Site"
-    url: "https://linktoyourblog123.com"
-    link_label: "linktoyourblog123.com"
-intro: "<p>Short introductory text</p>"
+  - label: "Personal site"
+    url: "https://kilianvalkhof.com"
+    link_label: "kilianvalkhof.com"
+  - label: "Polypane"
+    url: "https://polypane.app"
+    link_label: "polypane.app"
+intro: "<p>Specifing the right relationships between elements is a key part of building websites well. Often these relationships are implicit and depend on the source order or structure, such as a heading followed by a paragraph or a list element inside of a `<nav>`. And while this gets you far, there are times when you need to be more explicit about the relationships between elements.</p>"
 image: "advent25_5"
 ---
-Some text.
-Some text.
 
-Some text. Some text.
+Take the following HTML for example:
 
-## Heading
+```html
+<label>
+  Email address
+  <input type="email" />
+</label>
+```
+
+Here, the `<label>` element is associated with the `<input>` element because the input is nested inside of the label, and the browser now knows to offer additional behavior: when the user clicks on the label, the input will be focused. That's not behavior that you had to specify with an onclick handler, the browser just infers it from the relationship between the elements.
+
+But what if it can't infer the relationship? While your Firefox, Chrome or Polypane browser can all figure this out, some assistive technologies can't. Both Dragon Naturally Speaking and VoiceOver on macOS have trouble associating labels with inputs when they are nested, according to [a11ysupport tests](https://a11ysupport.io/tests/html_label_element_implicit) (Though support might have improved since the tests were last conducted). This means that while you clicking the label will work to focus the input, someone giving voice commands might not be able to say "Focus Email address" and have it work.
```

Comment:
```suggestion
But what if it can't infer the relationship? While your Firefox, Chrome or Polypane browser can all figure this out, some assistive technologies can't. Both Dragon Naturally Speaking and VoiceOver on macOS have trouble associating labels with inputs when they are nested, according to [a11ysupport tests](https://a11ysupport.io/tests/html_label_element_implicit) (Though support might have improved since the tests were last conducted). This means that while clicking the label will work to focus the input, someone giving voice commands might not be able to say "Focus Email address" and have it work.
```

Reply:
I think it's worthwhile mentioning that its actually people saying something out loud, so I'll rewrite it into a combination of the two.

Reply:
that makes a lot of sense, thank you!